core/src/main/java/io/questdb/cairo/CairoException.java

```diff
@@ -39,9 +39,9 @@ public class CairoException extends RuntimeException implements Sinkable, Flywei
     private static final ThreadLocal<CairoException> tlException = new ThreadLocal<>(CairoException::new);
     private static final StackTraceElement[] EMPTY_STACK_TRACE = {};
     protected final StringSink message = new StringSink();
-    protected int errno;
+    protected int errno; // code -1 is used for non-critical errors
```

Comment:
Have you considered expressing this intent (-1 == non-critical) in the CairoException API?

Right now there is `instance(int errno)` and when it's non-critical then you have to know to pass -1. 
We could change split the `instance(int errno)` to:
1. `critical(int errno)`
2. `nonCritical()`
With this it should be simpler to keep even new code consistent.

Reply:
Good idea. Let me implement it.

Reply:
Tried to apply the suggestion in 6f4adf4ecf066d411cce9b5cbcb8932619d4584d